aiojobs/_job.py

```diff
@@ -50,6 +56,20 @@ def pending(self) -> bool:
     def closed(self) -> bool:
         return self._closed
 
+    def get_name(self) -> str:
+        name = self._name
+        if sys.version_info >= (3, 8) and self._task:
+            name = self._task.get_name()
+        if name is None:
+            # Return generated default name
+            name = f"Job({self._coro})"
+        return name
+
+    def set_name(self, name):
+        if sys.version_info >= (3, 8):
+            return self._task.set_name(name)
+        self._name = name
```

Comment:
```suggestion
        if sys.version_info >= (3, 8) and self._task is not None:
            self._task.set_name(name)
        else:
            self._name = name
```
This would error if `self._task` is `None` (probably one of the mypy errors).

If the task is not None, then we also don't need to set the attribute anymore.